include/zephyr/net/coap_service.h

```diff
@@ -0,0 +1,284 @@
+/*
+ * Copyright (c) 2023 Basalte bv
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file
+ * @brief CoAP Service API
+ *
+ * An API for applications to respond to CoAP requests
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_COAP_SERVICE_H_
+#define ZEPHYR_INCLUDE_NET_COAP_SERVICE_H_
+
+#include <zephyr/net/coap.h>
+#include <zephyr/sys/iterable_sections.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief CoAP Service API
+ * @defgroup coap_service CoAP service API
```

Comment:
I would add this to the API Reference section of the `connectivity/networking/api/coap`  page

Reply:
So, not creating a new group? It does depend on an additional `Kconfig` setting besides `CONFIG_COAP` though.

Reply:
I think a new group is fine, I was simply recommending the following so that it actually shows up in the Sphinx docs:

```diff
diff --git a/doc/connectivity/networking/api/coap.rst b/doc/connectivity/networking/api/coap.rst
index 4f0654a57d..eae20b1f36 100644
--- a/doc/connectivity/networking/api/coap.rst
+++ b/doc/connectivity/networking/api/coap.rst
@@ -204,3 +204,4 @@ API Reference
 *************

 .. doxygengroup:: coap
+.. doxygengroup:: coap_service
```

On the other hand, it looks like coap_client has its own documentation page, so I will let @rlubos and @jukkar comment on what they think is best.

Reply:
Sorry for the delay, I somehow missed the comment... I think that more elaborate documetnation (like we have for CoAP client) would be ideal, but I don't want to block the PR with this. At the very least we should however reference the APIs in the CoAP doc page, as @kartben suggests.

Reply:
I will add a documentation page with examples, either to this, or a follow-up PR. 
Currently CoAP secure `coaps://` is higher on my priority list, which can also be added to this PR or the next

Reply:
The secure part will take more time as DTLS servers currently only support 1 client, I've added a documentation page.